sudoku_solver/sudoku/src/Generator.hs

```diff
@@ -0,0 +1,57 @@
+module Generator (generateSolved, getEasySudoku, getMiddleSudoku, getHardSudoku) where
+
+import Board
+import Solver (findEmpty, isValid, solve)
+import SampleBoards (emptyBoard)
+
+import Data.Either
+
+import System.Random (mkStdGen, randomIO)
+import System.Random.Shuffle (shuffle')
+
+getEasySudoku :: IO Board
+getEasySudoku = do
+    board <- generateSolved
+    removeRandomly board 30
+
+getMiddleSudoku :: IO Board
+getMiddleSudoku = do
+    board <- generateSolved
+    removeRandomly board 40
+
+getHardSudoku :: IO Board
+getHardSudoku = do
+    board <- generateSolved
+    removeRandomly board 50
+
+randomPermutation :: Int -> [a] -> [a]
+randomPermutation seed xs = shuffle' xs (length xs) (mkStdGen seed)
+
+generateSolved :: IO Board
+generateSolved = fillRandomly emptyBoard
+
+headMay :: [a] -> Maybe a
+headMay [] = Nothing
+headMay (x:_) = Just x
```

Comment:
There's a library function for this

Reply:
I didn't want to add one more library to the dependencies just for one function that can't be implemented this easily. Is doing so makes makes sense?

Reply:
@petyb It's in base, so no new dependency needed.

Reply:
isn't it in safe library?

Reply:
> isn't it in safe library?

It's in multiple libraries, but it's also in `base`, in the `Data.Maybe` module. 

![Screenshot 2024-12-16 at 12 11 58](https://github.com/user-attachments/assets/9f633a60-0756-491f-bd46-82e433bdbf6d)